homeassistant/components/mqtt/tag.py

```diff
@@ -21,10 +23,19 @@
     send_discovery_done,
     update_device,
 )
-from .models import MqttValueTemplate, ReceiveMessage
+from .models import MqttValueTemplate, ReceiveMessage, ReceivePayloadType
 from .subscription import EntitySubscription
 from .util import get_mqtt_data, valid_subscribe_topic
 
+if TYPE_CHECKING:
+
+    class TagComponent:
+        """Dummy class for type checking on module homeassistant.components.tag."""
+
+        async def async_scan_tag(self, tag_id: str, device_id: str | None) -> None:
```

Comment:
In `homeassistant.components.tag` for `async_scan_tag` attribute `device_id` must be `str`.
But when asserting `device_id` not to be None some tests fail. May be this is a typing issue in `homeassistant.components.tag.async_scan` or the tests must be fixed if we want to assert `device_id`.

Reply:
@emontnemery Do you know why in the `async_scan_tag` is passing `None` for `device_id`. The typing on `homeassistant.components.tag.async_scan` indicates that `device_id` should not be `None`.

Reply:
Moved to https://github.com/home-assistant/core/pull/81495.